packages/components/button-group/src/button-group.css

```diff
@@ -4,3 +4,16 @@
 	display: block;
 	width: 100%;
 }
+
+:host([inline]) {
+	font-size: 0;
+}
+
+:host([inline]) ::slotted(ts-button) {
+	margin-right: var(--ts-unit-half);
```

Comment:
Did you try `margin-inline-end: var(--ts-unit-half);`? The result looks the same for me, and it allows to drop the css rule below ( the one for rtl case).